### PR TITLE
Update README's for the 0.4.0 release

### PR DESCRIPTION
**Pull request by allisonport-db**
- Move the Hive docs to `hive/README.md`
- Add a `Flink/Delta Connector` section to the main readme
- Misc. changes including...
    - Using "Flink/Delta Connector" naming
    - Adding link to (future) Java docs for the Flink connector
    - some formatting fixes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,133 +25,20 @@ See [Delta Standalone](https://docs.delta.io/latest/delta-standalone.html) for d
...
-
-#### What if I change the underlying Delta table schema in Spark after creating the Hive table?
-If the schema in the underlying Delta metadata is not consistent with the schema specified by `CREATE TABLE` statement, the connector will report an error when loading the table and ask you to fix the schema. You must drop the table and recreate it using the new schema. Hive 3.x exposes a new API to allow a data source to hook ALTER TABLE. You will be able to use ALTER TABLE to update a table schema when the connector supports Hive 3.x.
-
-#### Hive has three execution engines, MapReduce, Tez and Spark. Which one does this connector support?
-The connector supports MapReduce and Tez. It doesn't support Spark execution engine in Hive.
+Use the [Flink/Delta Connector](flink-connector/README.md) to read and write Delta tables from Apache Flink applications. The connector includes a sink for writing to Delta tables from Apache Flink, and a source for reading Delta tables into Apache Flink (still in progress.) See the dedicated [README.md](/flink-connector/README.md) for more details.
```
Comment (scottsand-db) [inline README.md:34]: `source for reading Delta tables using Apache Flink (still in progress). See`

File: flink-connector/README.md [hunk 2]
```diff
@@ -1,39 +1,42 @@
...
-  able to read from it
-- Flink DeltaSink provides exactly-once delivery guarantees
-- depending on the version of the connector you can use it with following Apache Flink versions:
+- Currently only `DeltaSink` is supported, and thus the connector supports writing to Delta tables, but does not support reading Delta tables.
+- `DeltaSink` provides exactly-once delivery guarantees.
+- Depending on the version of the connector you can use it with following Apache Flink versions:
   
   | connector's version  | Flink's version |
   | :---: | :---: |
   |    0.2.1-SNAPSHOT    |    >= 1.12.0    |
```
Comment (scottsand-db) [inline flink-connector/README.md:23]: can we remove the `SNAPSHOT`? it shouldn't be there in the first place tbh

Comment (scottsand-db) [reply flink-connector/README.md:23]: and the next release is 0.4.0 ... and that is the first version to support the flink connector ... so let's make this `0.4.0`. what do you think?

**Review by scottsand-db — APPROVED** [review]
LGTM with a few minor suggestions. Thanks!

File: flink-connector/README.md (hunk 2 above)
Comment (allisonport-db) [reply flink-connector/README.md]: Definitely agree

**Pull request merged**